packages/wrangler/src/versions/upload.ts

```diff
@@ -702,7 +727,10 @@ See https://developers.cloudflare.com/workers/platform/compatibility-dates for m
 			compatibility_date: compatibilityDate,
 			compatibility_flags: compatibilityFlags,
 			keepVars: props.keepVars ?? false,
-			keepSecrets: true, // until wrangler.toml specifies secret bindings, we need to inherit from the previous Worker Version
+			keepSecrets: props.secretsFile ? false : true,
+			keepBindings: props.secretsFile
+				? ["secret_key", "secret_text"]
+				: undefined,
```

Comment:
I don't think this change is quite right. They cancel each other out.

When `keepSecrets` is true (i.e. here `secretsFile` is falsy) we automatically add `secret_key` and `secret_text` to the keep_bindings field in `createWorkerUploadForm()`.

And here if `secretsFile` is truthy, then we are explicitly adding `secret_key` and `secret_text`.

If we are saying that `secretsFile` is the full source of truth for secrets then we should be setting `keepSecrets` to false and not setting `keepBindings` at all.

If we are saying that `secretsFiles` is additive to the current secrets in the dashboard then `keepSecrets` should always be true and `keepBindings` should not be defined.

From looking at how `wrangler versions secret bulk` works, we should not be treating the secetsFile as a source of truth but only additive.  I think we should follow that here too.

So I think this should be:


```suggestion
			// we never delete secret bindings when uploading, even if we are setting secrets from a file
			// so inherit all unchanged secrets from the previous Worker Version
			keepSecrets: true, 
```